synapse/handlers/federation_event.py

```diff
@@ -955,6 +955,12 @@ async def _compute_event_context_with_maybe_missing_prevs(
         seen = await self._store.have_events_in_timeline(prevs)
         missing_prevs = prevs - seen
 
+        # Filter out events we've tried to pull recently
+        prevs_to_ignore = await self.store.filter_events_with_pull_attempt_backoff(
+            room_id, missing_prevs
+        )
+        missing_prevs = missing_prevs - prevs_to_ignore
```

Comment:
Maybe we should just bail early if there are any `prevs_to_ignore` since

https://github.com/matrix-org/synapse/blob/958fd3b5edb69e4bc65ec2679bbe3f305713bdf2/synapse/handlers/federation_event.py#L948-L949